internal/ignition/boot-reporter/assisted-boot-reporter.sh

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env bash
+
+export LOG_SEND_FREQUENCY_IN_MINUTES=5
+export SERVICE_TIMEOUT_MINUTES=60
+
+function log() {
+  echo "$(date '+%F %T') ${HOSTNAME} $2[$$]: level=$1 msg=\"$3\""
+}
+
+function log_info() {
+  log "info" "$1" "$2"
+}
+
+function log_error() {
+  log "error" "$1" "$2"
+}
+
+function init_variables() {
+  func_name=${FUNCNAME[0]}
+  init_failed="false"
+
+  if [ "$LOG_SEND_FREQUENCY_IN_MINUTES" == "" ]; then
+    init_failed="true"
+    log_error "${func_name}" "LOG_SEND_FREQUENCY_IN_MINUTES is empty."
+  fi
+
+  if [ "$SERVICE_TIMEOUT_MINUTES" == "" ]; then
+    init_failed="true"
+    log_error "${func_name}" "SERVICE_TIMEOUT_MINUTES is empty."
+  fi
+
+  if [ "$ASSISTED_SERVICE_URL" == "" ]; then
+    init_failed="true"
+    log_error "${func_name}" "ASSISTED_SERVICE_URL is empty."
+  elif [ "${ASSISTED_SERVICE_URL: -1}" == "/" ]; then
+    export ASSISTED_SERVICE_URL="${ASSISTED_SERVICE_URL::-1}"
+  fi
+
+  if [ "$CLUSTER_ID" == "" ]; then
+    init_failed="true"
+    log_error "${func_name}" "CLUSTER_ID is empty."
+  fi
+
+  if [ "$INFRA_ENV_ID" == "" ]; then
+    init_failed="true"
+    log_error "${func_name}" "INFRA_ENV_ID is empty."
+  fi
+
+  if [ "$HOST_ID" == "" ]; then
+    init_failed="true"
+    log_error "${func_name}" "HOST_ID is empty."
+  fi
+
+  if [ "$init_failed" == "true" ]; then
+    log_error "${func_name}" "Failed to initialize variables. Exiting."
+    exit 1
+  fi
+}
+
+function collect_and_upload_logs() {
+  func_name=${FUNCNAME[0]}
+
+  log_info "${func_name}" "Collecting logs."
+  logs_dir_name=boot_logs_$HOST_ID
+  logs_path=/tmp/$logs_dir_name
+
+  rm -rf $logs_path
+  mkdir -p $logs_path
+
+  journalctl > "$logs_path"/journalctl.log
+  log_info "${func_name}" "Copying journalctl to $logs_path/"
+  ip a > $logs_path/ip_a.log
+  log_info "${func_name}" "Capturing the output of 'ip a' to $logs_path/"
+  cp /etc/resolv.conf $logs_path
+  log_info "${func_name}" "Copying /etc/resolv.conf to $logs_path/"
+
+  pushd /tmp
+  log_info "${func_name}" "Compressing logs to $logs_dir_name.tar.gz"
+  tar -czvf "$logs_dir_name".tar.gz "$logs_dir_name"
+  popd
+
+  log_info "${func_name}" "Uploading logs."
+
+  curl -X POST  -H "X-Secret-Key: ${PULL_SECRET_TOKEN}" \
```

Comment:
According to the swagger you don't need the infra_env_id in the query params when POSTing logs.
https://github.com/openshift/assisted-service/blob/64897a310a4b5f7eae7d115e159e8f7303b1ceab/swagger.yaml#L3084
By removing the infra_env_id you can avoid the db addition to the generator struct, less code changes +  reduce DB calls.

Reply:
That is a fair point. Indeed it should not be a must. I'll test it without `infra_env_id`.
https://github.com/openshift/assisted-service/blob/64897a310a4b5f7eae7d115e159e8f7303b1ceab/swagger.yaml#L3173-L3178

Reply:
> That is a fair point. Indeed it should not be a must. I'll test it without `infra_env_id`.
> 
> https://github.com/openshift/assisted-service/blob/64897a310a4b5f7eae7d115e159e8f7303b1ceab/swagger.yaml#L3173-L3178

/hold
testing this in CI.

Reply:
I checked it, and it looks like infra_env should be marked as `required: true`.
Looking at:
https://github.com/openshift/assisted-service/blob/8b8e06e2eaf14d6497364d7fc0d4697d256cfe89/internal/bminventory/inventory_v2_handlers.go#L463-L487

We use `infra_env_id` to fetch the host from DB.
Even if I remove it, we also use it down the line with events.

This was the case for the existing host logs before I added node-boot logs, so this is not new.

Reply:
Based on test results, it will force users to specify infra_env_id even when they upload cluster logs. I'll revert back to `required: false`. It looks out of scope for this PR.

Reply:
done

Reply:
I see, sorry for all the truble this put you through

Reply:
Note that you could update:
`dbHost, err := common.GetHostFromDB(b.db, params.InfraEnvID.String(), params.HostID.String()` 
to:
`dbHost, err := common.GetHostFromDBByHostId(b.db, params.InfraEnvID.String(), params.HostID.String()`

Reply:
Sec, i agree with @eranco74 actually, if you don't need infraenv why we need to add  it and complicate so many stuff.
There is no  need for it at all. 
If you not must use it please remove